neo.UnitTests/Network/RPC/Server/UT_RpcServer.cs

```diff
@@ -0,0 +1,168 @@
+            {
+                int.TryParse(b, out var bInt);
+                return bInt.Equals(a);
+            }));
+
+            rpcServer.BindController<MathController>();
+            rpcServer.BindController(typeof(Foo));
+
+            // inject
+
+            rpcServer.InjectSpecialParameter(context => new RandomObjectForTest { RandomProp = "Hello" });
+
+            // calling
+
+            var resp1 = (bool) rpcServer.CallOperation(null, null, "checkDifferent", 2, "2");
+            Assert.IsFalse(resp1);
+
+            var resp2 = (bool) rpcServer.CallOperation(null, "math", "checkEquals", new Dictionary<string, object>
+            {
+                {"a", 2 },
+                {"b", "2"}
+            });
+            Assert.IsTrue(resp2);
+
+            var resp3 = (int) rpcServer.CallOperation(null, "math", "sum", 2, 3);
+            Assert.AreEqual(resp3, 5);
+
+            var resp4 = (int) rpcServer.CallOperation(null, "math", "sub", 3, 1);
+            Assert.AreEqual(resp4, 2);
+
+            var resp5 = (string) rpcServer.CallOperation(null, "Foo", "Bar");
+            Assert.AreEqual(resp5, "foo bar :)");
+
+            var resp6 = (string) rpcServer.CallOperation(null, "Foo", "PrettyMessage", "how are you?");
+            Assert.AreEqual(resp6, "Hello, how are you?");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void unbind()
+        {
+            RpcServer rpcServer = new RpcServer(GetRpcConfig());
+
+            // binding
+
+            rpcServer.BindOperation(null, "checkDifferent", new Func<int, string, bool>((a, b) =>
+            {
+                int.TryParse(b, out var bInt);
+                return !bInt.Equals(a);
+            }));
+            rpcServer.BindController<Foo>();
+
+            // unbinding
+
+            rpcServer.UnbindController("Foo");
+
+            // calling
+
+            var resp = (bool) rpcServer.CallOperation(null, null, "checkDifferent", 2, "2");
+            Assert.IsFalse(resp);
+
+            rpcServer.CallOperation(null, "Foo", "Bar");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void unbindAll()
+        {
+            RpcServer rpcServer = new RpcServer(GetRpcConfig());
+
+            // binding
+
+            rpcServer.BindOperation(null, "checkDifferent", new Func<int, string, bool>((a, b) =>
+            {
+                int.TryParse(b, out var bInt);
+                return !bInt.Equals(a);
+            }));
+
+            // unbinding
+
+            rpcServer.UnbindAllOperations();
+
+            // calling
+
+            rpcServer.CallOperation(null, null, "checkDifferent", 2, "2");
+        }
+
+        private static RpcConfig GetRpcConfig()
+        {
+            return new RpcConfig
+            {
+                ListenEndPoint = new IPEndPoint(IPAddress.Parse("127.0.0.1"), 10322),
```

Comment:
Will work these tests in parallel? I don't think so, will be great if all test use a different port, i want to parallelize all the unit test in a future.

Reply:
Done